requests-for-comments/2022-11-08-supported-languages.md

```diff
@@ -0,0 +1,169 @@
+# Request for comments: Supported languages at PostHog
+
+Since day 1 here at Posthog we've supported two and half core languages: Python and Javascript/Typescript.
+
+This set of tools has carried us a long way and we should both the cpython runtime and node a huge hats off for bringing us this far.
+
+## Problem statement
+
+We have the opportunity coming up to rebuild or greenfield build out services that are critical parts of our data pipeline. It will be important for these services to be correct, fast, and efficient. Considering this now is a good time to ask: Are we using the correct tools for the job?
+
+So why even bring this up?
+
+Frankly:
+
+- Python is slow.
+- Node is a memory hog.
+- Node tooling is slow.
+- Dependencies can be huge.
+- No guarantees that code is correct.
```

Comment:
> - Python is slow.

True generally, but relevant only for specific usecases (like here).

> - Node is a memory hog.

Has this been a problem? There are alternative implementations of JS (e.g. [bun](https://bun.sh/), [deno](https://deno.land/)) that do this better, but are too fresh to rely on in production.

> - Node tooling is slow.

Depends on the tooling. The frontend builds quite fast (sub 3-10sec), the plugin server can be improved 10x if needed.

> - Dependencies can be huge.

We're [switching to `pnpm`](https://github.com/PostHog/posthog/pull/12635) that'll fix this

> - No guarantees that code is correct.

Can you ellaborate?